jme3-terrain/src/main/java/com/jme3/terrain/geomipmap/TerrainPatch.java

```diff
@@ -255,9 +255,9 @@ protected void reIndexGeometry(HashMap<String,UpdatedTerrainPatch> updated, bool
 
             Buffer b;
             if (idxB.getBuffer() instanceof IntBuffer)
-                b = (IntBuffer)idxB.getBuffer();
+                b = idxB.getBuffer();
             else
-                b = (ShortBuffer)idxB.getBuffer();
+                b = idxB.getBuffer();

```

Comment:
Same here. Could someone try to prevent Byte and Long Buffers on purpose? Or is this just a mistake?

Reply:
I'll remove the test. Probably when jme3-terrain was written, byte-sized indices were rare or even unsupported. I'm pretty sure JME still doesn't support long-sized indices.